DSP/APCompressor.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+  ==============================================================================
+
+    APCompressor.cpp
+    Created: 26 Feb 2021 2:08:02pm
+    Author:  Johnathan Handy
+
+  ==============================================================================
+*/
+
+#include "APCompressor.h"
+#include "JuceHeader.h"
+
+APCompressor::APCompressor()
+{
+
+}
+
+APCompressor::~APCompressor()
+{
+
+}
+
+void APCompressor::process (float* audioIn,
+                            float* audioOut,
+                            float numSamplesToRender)
+{
+    for (auto i = 0; i < numSamplesToRender; ++i)
+    {
+        audioOut[i] = applyRMSCompression (audioIn[i]);
+    }
+}
+
+float APCompressor::applyFFCompression (float sample)
+{
+    auto alphaA = exp(-log(9) / (sampleRate_ * attack_));
+    auto alphaR = exp(-log(9) / (sampleRate_ * release_));
+
+    float gainSmooth = 0;
+    float gain_sc = 0;
+
+    auto x_uni = abs(sample);
+    auto x_dB = 20 * log10(x_uni);
+    if (x_dB < MIN_DB)
+        x_dB = MIN_DB;
```

Comment:
Should use floats, and can do as single expression instead of requiring `if`, allows use of `const`

```suggestion
    const auto x_dB = juce::jmax(20.f * log10(x_uni), MIN_DB);
```